### PR TITLE
Fix test_learner_with_slow_snapshot test case

### PR DESCRIPTION
**Pull request by hicqu**
## What have you changed? (mandatory)
- improve a test case.

## What are the type of the changes? (mandatory)
bugfix.

## How has this PR been tested? (mandatory)
make dev

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)
no

## Does this PR affect tidb-ansible update? (mandatory)
no

### CODE REVIEW

File: tests/raftstore_cases/test_conf_change.rs [hunk 1]
```diff
@@ -716,7 +716,10 @@ fn test_learner_with_slow_snapshot() {
     impl Filter<RaftMessage> for SnapshotFilter {
         fn before(&self, msgs: &mut Vec<RaftMessage>) -> Result<()> {
             let count = msgs.iter()
-                .filter(|m| m.get_message().get_msg_type() == MessageType::MsgSnapshot)
+                .filter(|m| {
+                    m.get_message().get_msg_type() == MessageType::MsgSnapshot
+                        && m.get_message().get_snapshot().has_metadata()
```
Comment (siddontang) [inline tests/raftstore_cases/test_conf_change.rs]: why this can improve the test?

**Comment by overvenus** [comment]
Could you rename the PR title?

**Pull request merged**